### PR TITLE
avoid copies by using references in ranged-for loops

### PR DESCRIPTION
**Pull request by froydnj**
One gotcha in C++ is that the convenient `for (auto x : iterable_thing)` expands to something that copies elements of `iterable_thing` into a variable named `x` for each iteration of the loop.  This is not so bad when you're iterating over a `std::vector<int32_t>`, but it results in a bunch of needless copies when iterating over a `std::vector<std::string>` or similar.  (If you are writing loops where you depend on the copying behavior...yuck.)

There is a [`clang-tidy` check for this footgun](https://clang.llvm.org/extra/clang-tidy/checks/performance-for-range-copy.html), but as we don't currently have `clang-tidy` enabled, I fixed a few cases manually.  I also fixed `Proto::toProto` to not copy its `typ` argument unnecessarily.  I did not exhaustively examine every eligible ranged-for loop because I didn't think it was worth it.

We could make a very limited linter for this sort of thing (match on the regex `for \(auto [A-Za-z0-9_]+ :` and disallow it), but that'd be far too restrictive: there are cases where you don't care if the value is copied.

### Motivation

Epsilon faster performance due to less copying.

### Test plan

This is not a user-visible change, and the existing tests should be sufficient.

### CODE REVIEW

File: core/Symbols.cc [hunk 1]
```diff
@@ -32,7 +32,7 @@ bool SymbolRef::operator!=(const SymbolRef &rhs) const {
 vector<TypePtr> Symbol::selfTypeArgs(const GlobalState &gs) const {
     ENFORCE(isClassOrModule()); // should be removed when we have generic methods
     vector<TypePtr> targs;
-    for (auto tm : typeMembers()) {
+    for (auto &tm : typeMembers()) {
```
Comment (jez) [inline core/Symbols.cc:35]: `SymbolRef` is cheap to copy

Comment (jez) [reply core/Symbols.cc:35]:
```suggestion
    for (auto tm : typeMembers()) {
```

**Review by jez — APPROVED** [review]
One suggestion. Everything else looks good to me.

File: core/Symbols.cc (hunk 1 above)
Comment (froydnj) [reply core/Symbols.cc:35]: Bah, somehow I think I determined that `typeMembers()` was returning a vector of `TypePtr`; Code led me astray!

Comment (jez) [reply core/Symbols.cc:35]: clangd is A+ here, it told me the type.

**Pull request merged**